client.js

```diff
@@ -280,6 +364,51 @@ class Client {
     return flags.join(" + ");
   }
 
+  /**
+ * Returns a human‐readable string for a given event code.
+ *
+ * @param {number} code - The numeric event code.
+ * @returns {string} - The corresponding event code string.
+ */
+  getEventCodeString(code) {
+    // Return empty string if eventCode is zero
+    if (code === 0) return "";
+
+    // Define the flag values (adjust these constants if needed)
+    const EventCodeFlags = {
+      AlarmSet: 0x1,
+      AlarmClr: 0x2,
+      AlarmAck: 0x4,
+      AlarmReprise: 0x40
+    };
+
+    // Check for specific combinations first
+    if (code === EventCodeFlags.AlarmSet) return "AlarmSet";
+    if (code === EventCodeFlags.AlarmClr) return "AlarmClear";
+    if (code === EventCodeFlags.AlarmAck) return "Ack";
+    if (code === EventCodeFlags.AlarmReprise) return "Reprise";
+    if (code === (EventCodeFlags.AlarmReprise | EventCodeFlags.AlarmSet))
+      return "RepriseAlarmSet";
+    if (code === (EventCodeFlags.AlarmReprise | EventCodeFlags.AlarmClr))
+      return "RepriseAlarmClear";
+    if (code === (EventCodeFlags.AlarmReprise | EventCodeFlags.AlarmAck))
+      return "RepriseAck";
+
+    // Otherwise, combine the flag strings based on which bits are set.
+    let s = "";
+    if (code & EventCodeFlags.AlarmReprise)
+      s += (s ? "+" : "") + "Reprise";
+    if (code & EventCodeFlags.AlarmSet)
+      s += (s ? "+" : "") + "AlarmSet";
+    if (code & EventCodeFlags.AlarmClr)
+      s += (s ? "+" : "") + "AlarmClear";
+    if (code & EventCodeFlags.AlarmAck)
+      s += (s ? "+" : "") + "Ack";
```

Comment:
The C++ code I sent also handled NodeBoot and SourceObjectUnavailable. Should still include them